web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java

```diff
@@ -79,12 +79,18 @@ static AuthTokenPromiseSupplier oneShot(ConnectToken initialToken) {
     private boolean connected;
     private boolean closed;
     private boolean hasDisconnected;
+    private boolean notifiedConnectionError = false;
 
     public QueryConnectable(Supplier<Promise<ConnectToken>> authTokenPromiseSupplier) {
         this.connection = JsLazy.of(() -> new WorkerConnection(this, authTokenPromiseSupplier));
     }
 
     public void notifyConnectionError(ResponseStreamWrapper.Status status) {
+        if (notifiedConnectionError) {
```

Comment:
What's the threading model here? Should this be an `AtomicBoolean`, or should the method/block be synchronized?

Reply:
In javascript the threading model is ... you get one thread. That's it! That's the main reason it is extremely important to design all browser-executing code as async as possible.